[YuZizhen] iP

YuZizhen opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

YuZizhen committed:
Level-0

YuZizhen committed:
Level-1

YuZizhen committed:
Level-2

YuZizhen committed:
Add Task.java

YuZizhen committed:
Level-3

YuZizhen committed:
Level-4

YuZizhen committed:
Level-5

YuZizhen committed:
Level-6

YuZizhen committed:
Create git l

YuZizhen committed:
A-TextUiTextCompare

YuZizhen committed:
Level-7

YuZizhen committed:
Level-8

Convert date to LocalDate

YuZizhen committed:
A-MoreOOP

YuZizhen committed:
A-Packages

YuZizhen committed:
Merge remote-tracking branch 'origin/add-gradle-support'

YuZizhen committed:
JUnit Test for Parser

YuZizhen committed:
A-JUnit

YuZizhen committed:
A-Jar

YuZizhen committed:
add some javadoc and classdoc to the method and fucntions

YuZizhen committed:
A-JavaDoc

YuZizhen committed:
A-CodingStandard

YuZizhen committed:
Level-9

YuZizhen committed:
Merge branch 'branch-A-CodingStandard'

YuZizhen committed:
Merge branch 'branch-Level-9'

tanyunchao commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,29 @@
+        duke.greeting();
+
+        ArrayList<Task> temp = new ArrayList<>();
+        temp = storage.getHistory();
```
Nice way to name the method! Very clear what it does without even looking into the code itself.

tanyunchao commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+     * @param userInput the input from the user.
+     */
+
+    public void echo(String userInput) {
```
hmm its not immediately obvious to me what this echo method is supposed to do at a glance. Only by reading the javadocs was I able to understand it's supposed to handle the inputs routing. Maybe we could consider using a different name for the method?

jasperng-nus replied:
I think the name of this method could be misleading as I also did not understand what it means just by looking at it, perhaps you could change the name to, handleTaskInput?

tanyunchao commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+     * @return True means the user want to delete, or false, which is otherwise.
+     */
+
+    public boolean isDeleteTask(String userInput) {
```
Good naming of the methods, very clear to anyone thats reading the code that the function returns a boolean!

jasperng-nus replied:
LGTM, The naming is clear and I could understand that this method is to parse the task that contain delete just by looking at it, good job!

tanyunchao commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+package task;
```
correct casing used for package declaration!

tanyunchao reviewed (commented):
Hi Zizhen, overall very neat and readable code! It was very easy to review for you haha. One thing you might want to consider changing the `Duke.java` file to be the name of your chatbot instead.

jasperng-nus commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+            } else if (isFindTask(userInput)) {
+                this.todoList.findTask(userInput, storage);
+            } else {
+                this.echo(userInput);
```
LGTM. I like the way you handle the inputs, and having different methods to handle the different tasks. It looks neat and the names are well understood, I could tell what it does just by looking at it.

jasperng-nus commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+     * @return True means the user want to mark, or false, which is otherwise.
+     */
+
+    public boolean isMarkTask(String userInput) {
```
LGTM. The naming is clear and I could understand that this method is to parse the task that contains mark just by looking at it, good job!

jasperng-nus commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+        return false;
+    }
+
+    public boolean isFindTask(String userInput) {
```
LGTM. Same goes to this!

jasperng-nus commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+        }
+    }
+
+    private void showResult(TaskList result, String keyword) {
```
LGTM. Good use of names that clearly shows what the method does just by looking at it.

jasperng-nus reviewed (commented):
Overall good job! keep up the good work! just some minor feedback i gave, you could see if it helps!

YuZizhen committed:
A-CheckStyle

YuZizhen committed:
Simple javafx implemented

YuZizhen committed:
DialogBox implemented

YuZizhen committed:
Level-10

YuZizhen committed:
A-Assertions

Improve on the potential problem

Task should have a non-empty description. Using assertions can potentially avoid such problem.

YuZizhen committed:
A-CodeQuality

Refactor some of the methods to make the code neater.

YuZizhen committed:
A-CI

YuZizhen committed:
Add archive extension

Add archive extension methods

These methods allows the deleted task to be put under the archived file.

It will remain a copy of the deleted tasks and improves user experience.

YuZizhen committed:
A-UserGuide

YuZizhen committed:
Change to String output

YuZizhen committed:
C-Archive

YuZizhen committed:
Merge branch 'branch-archiveExtension'

YuZizhen committed:
A-Personality

YuZizhen committed:
More description added

YuZizhen committed:
Update README.md

YuZizhen committed:
Fix exit function

YuZizhen committed:
Merge branch 'master' of https://github.com/YuZizhen/ip

YuZizhen committed:
Add screen shot of the application.

YuZizhen committed:
Update user guide.

YuZizhen committed:
Update main

YuZizhen committed:
Update description of Storage

YuZizhen committed:
Update description of Exit

YuZizhen committed:
Update user guide for Archive

YuZizhen committed:
Update description of Parser

YuZizhen committed:
Merge branch 'master' into branch-archiveExtension

YuZizhen committed:
Merge pull request #2 from YuZizhen/branch-archiveExtension

Update user guide

YuZizhen committed:
Update README.md
